File: python/examples/errors/basic.py
from mirascope import llm


@llm.call("openai/gpt-5-mini")
def recommend_book(genre: str):
    return f"Recommend a {genre} book."


try:
    response = recommend_book("fantasy")
    print(response.pretty())
except llm.AuthenticationError as e:
    print(f"Invalid API key: {e}")
except llm.RateLimitError as e:
    print(f"Rate limit exceeded: {e}")
except llm.Error as e:
    print(f"LLM error: {e}")

File: typescript/examples/errors/basic.ts
/**
 * Basic error handling example.
 *
 * Demonstrates catching and handling common LLM errors.
 *
 * Run with: bun run example examples/errors/basic.ts
 */
import { llm } from "mirascope";

const recommendBook = llm.defineCall<{ genre: string }>({
  model: "openai/gpt-4o-mini",
  template: ({ genre }) => `Recommend a ${genre} book.`,
});

try {
  const response = await recommendBook({ genre: "fantasy" });
  console.log(response.text());
} catch (e) {
  if (e instanceof llm.AuthenticationError) {
    console.log(`Invalid API key: ${e.message}`);
  } else if (e instanceof llm.RateLimitError) {
    console.log(`Rate limit exceeded: ${e.message}`);
  } else if (e instanceof llm.MirascopeError) {
    console.log(`LLM error: ${e.message}`);
  } else {
    throw e;
  }
}
